tests/misc/programs/misc/src/context.rs

```diff
@@ -336,3 +336,10 @@ pub struct TestConstArraySize<'info> {
     #[account(zero)]
     pub data: Account<'info, DataConstArraySize>,
 }
+
+#[derive(Accounts)]
+#[instruction(bump: u8)]
+pub struct TestProgramIdConstraint<'info> {
```

Comment:
can we test that this also works with the globally defined id()? I don't think that check is every enforced? @armaniferrante

Reply:
What check would you expect? seeds don't check the global ID. That is the job of the account types.

Reply:
Nvm, I completely misread the variable name, I thought the previously named variable `program_id` represents the program that the account belongs to and it checks that, rather than being generated from that seed.